common/containers/Tabs/BroadcastTx/index.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { AppState } from 'reducers';
+import TabSection from 'containers/TabSection';
+import { translateRaw } from 'translations';
+import { broadcastTx as dBroadcastTx, TBroadcastTx } from 'actions/wallet';
+import { QRCode } from 'components/ui';
+import './index.scss';
+import {
+  BroadcastTransactionStatus,
+  getTransactionFields
+} from 'libs/transaction';
+import EthTx from 'ethereumjs-tx';
+import { ConfirmationModal } from 'containers/Tabs/SendTransaction/components';
+
+interface Props {
+  broadcastTx: TBroadcastTx;
+  transactions: BroadcastTransactionStatus[];
+}
+
+interface State {
+  signedTx: string;
+  showConfirmationModal: boolean;
+  disabled: boolean;
+}
+
+const initialState: State = {
+  showConfirmationModal: false,
+  signedTx: '',
+  disabled: true
+};
+
+class BroadcastTx extends Component<Props, State> {
+  public state = initialState;
+
+  public ensureValidSignedTxInputOnUpdate() {
+    try {
+      const tx = new EthTx(this.state.signedTx);
+      getTransactionFields(tx);
+      if (this.state.disabled) {
+        this.setState({ disabled: false });
+      }
+    } catch (e) {
```

Comment:
TS lets you do `} catch { ..rest of code here` if you don't do anything with the error param